cancel_token/token.py

```diff
@@ -132,21 +143,35 @@ def cancel_not_done(fut: 'asyncio.Future[None]') -> None:
                 return_when=asyncio.FIRST_COMPLETED,
                 loop=self.loop,
             )
-        except asyncio.futures.CancelledError:
-            # Since we use return_when=asyncio.FIRST_COMPLETED above, we can be sure none of our
-            # futures will be done here, so we don't need to check if any is done before cancelling.
+        except asyncio.CancelledError as err:
+            # Since we use return_when=asyncio.FIRST_COMPLETED above, we can
+            # be sure none of our futures will be done here, so we don't need
+            # to check if any is done before cancelling.
             for future in futures:
                 future.cancel()
-            raise
-        for task in pending:
-            task.cancel()
+            for future in futures:
+                try:
+                    await future
+                except asyncio.CancelledError:
+                    pass
```

Comment:
3. This one fixes futures that aren't awaited after the `asyncio.wait()` is cancelled

Reply:
This is now covered in `test_awaitables_are_cancelled_and_cleaned_up_on_outer_cancellation`